site/src/Librecores/ProjectRepoBundle/RepoCrawler/GitRepoCrawler.php

```diff
@@ -2,14 +2,24 @@
 
 namespace Librecores\ProjectRepoBundle\RepoCrawler;
 
+use Doctrine\Common\Persistence\ObjectManager;
 use Librecores\ProjectRepoBundle\Entity\Commit;
+use Librecores\ProjectRepoBundle\Entity\Contributor;
 use Librecores\ProjectRepoBundle\Entity\GitSourceRepo;
 use Librecores\ProjectRepoBundle\Entity\LanguageStat;
 use Librecores\ProjectRepoBundle\Entity\ProjectRelease;
+use Librecores\ProjectRepoBundle\Entity\SourceRepo;
 use Librecores\ProjectRepoBundle\Util\FileUtil;
+use Librecores\ProjectRepoBundle\Util\MarkupToHtmlConverter;
+use Librecores\ProjectRepoBundle\Util\ProcessCreator;
+use Psr\Log\LoggerInterface;
 use Symfony\Component\Process\Process;
 use Symfony\Component\Filesystem\Filesystem;
 use Symfony\Component\Filesystem\Exception\IOExceptionInterface;
+use Librecores\ProjectRepoBundle\Entity\Project;
+use Librecores\ProjectRepoBundle\Util\StatsUtil;
+use Librecores\ProjectRepoBundle\Util\Dates;
+use Librecores\ProjectRepoBundle\Doctrine\ProjectMetricsProvider;
```

Comment:
I don't see code that makes use of all those added classes. Why are they necessary?

Reply:
For the constructor